generate_metap_agg.py

```
@@ -1,193 +1,440 @@
-        for actrec in self.act_col.find({"type":"nmdc:MetaproteomicsAnalysis"}):
-            # New annotations should have this
-            act_id = actrec['id']
-            # Skip if already processed
-            if act_id in done:
-                continue
-            try:
-                kegg_term_counts = self.process_activity(actrec)
-                rows = kegg_term_counts.values()
+        # Get list of workflow IDs that have already been processed
+        mp_wf_in_agg = self.get_previously_aggregated_workflow_ids()
+
+        # Get list of all workflow records
+        mp_wf_recs = self.get_workflow_records()
 
-                if DEBUG_MODE:
-                    print(f"{type(rows)=}")  # print type
-                    print(f"{len(rows)=}")  # print length
-                    print(f"{rows=}")  # print the value
+        # Records to add to the aggregation
+        agg_records = {}
 
+        # Iterate through all of the workflow records
+        for mp_wf_rec in mp_wf_recs:
+            if mp_wf_rec["id"] in mp_wf_in_agg:
+                continue
+            try:
+                agg_records[mp_wf_rec["id"]] = self.process_activity(mp_wf_rec)
             except Exception as ex:
                 # Continue on errors
                 print(ex)
                 continue
-            if len(rows) > 0:
-                print(f"Inserting: {len(rows)} rows for {act_id}")
-                self.agg_col.insert_many(rows)
+
+            # Prepare a  json record for the database
+            json_records = []
+            for key, value in agg_records.items():
+                for k, v in value.items():
+                    json_records.append(
+                        {"was_generated_by": key, "gene_function_id": k, "count": v, "type": "nmdc:FunctionalAnnotationAggMember"}
+                    )
+            json_record_full = {"functional_annotation_agg": json_records}
+
+            response = self.submit_json_records(json_record_full)
```

This will error out with an Exception rather than logging an error and moving to the next record if you don't succeed in the submit call. That may be the correct behavior (since it is likely a bad server connection or similar), but just wanted to make sure.